src/Pool.sol

```diff
@@ -56,6 +66,15 @@ abstract contract Pool is Pausable, Ownable, ERC4626 {
     /// @notice Variable to store the total amount deposited.
     uint128 private _totalAmount;
 
+    /// @notice Variable to store the recipient address that will receive the deposit fees.
+    address private _depositFeeRecipient;
+
+    /// @notice Variable to store the rewards per token.
+    uint128 private _rewardsPerToken;
+
+    /// @notice Variable to store the last block when the rewardsPerToken was updated.
+    uint128 private _lastBlockUpdated;
+
     /// @notice Array to store the Reward Multipliers.
     RewardsMultiplier[] private _rewardsMultipliers;
```

Comment:
gas: add a variable to keep up with the current reward multiplier and withdraw fee.